src/main/fc/rc.c

```diff
@@ -489,141 +492,177 @@ static FAST_CODE void processRcSmoothingFilter(void)
 }
 #endif // USE_RC_SMOOTHING_FILTER
 
-NOINLINE void initAveraging(uint16_t feedforwardAveraging)
-{
-    for (int i = 0; i < XYZ_AXIS_COUNT; i++) {
-        laggedMovingAverageInit(&feedforwardDeltaAvg[i].filter, feedforwardAveraging + 1, (float *)&feedforwardDeltaAvg[i].buf[0]);
-    }
-}
-
-FAST_CODE_NOINLINE void calculateFeedforward(const pidRuntime_t *pid, int axis)
+#ifdef USE_FEEDFORWARD
+FAST_CODE_NOINLINE void calculateFeedforward(const pidRuntime_t *pid, flight_dynamics_index_t axis)
 {
     const float rxInterval = currentRxIntervalUs * 1e-6f; // seconds
-    float rxRate = currentRxRateHz;
-    static float prevRxInterval;
-
-    static float prevRcCommand[3];
+    float rxRate = currentRxRateHz;                 // 1e6f / currentRxIntervalUs;
+    static float prevRcCommand[3];                  // for rcCommandDelta test
     static float prevRcCommandDeltaAbs[3];          // for duplicate interpolation
-    static float prevSetpoint[3];                   // equals raw unless interpolated 
-    static float prevSetpointSpeed[3];
-    static float prevAcceleration[3];               // for duplicate interpolation
-    static bool prevDuplicatePacket[3];             // to identify multiple identical packets
-    static uint16_t feedforwardAveraging = 0;
-
-    if (feedforwardAveraging != pid->feedforwardAveraging) {
-        feedforwardAveraging = pid->feedforwardAveraging;
-        initAveraging(feedforwardAveraging);
-    }
+    static float prevSetpoint[3];                   // equals raw unless extrapolated forward 
+    static float prevSetpointSpeed[3];              // for setpointDelta calculation
+    static float prevSetpointSpeedDelta[3];         // for duplicate extrapolation
+    static bool isPrevPacketDuplicate[3];             // to identify multiple identical packets
 
-    const float rcCommandDeltaAbs = fabsf(rcCommand[axis] - prevRcCommand[axis]);
+    const float rcCommandDelta = rcCommand[axis] - prevRcCommand[axis];
     prevRcCommand[axis] = rcCommand[axis];
+    float rcCommandDeltaAbs = fabsf(rcCommandDelta);
```

Comment:
7(Maybe use `fabsf(rcCommandDelta)` instead `rcCommandDeltaAbs` and keep `prevRcCommandDelta`)
`fabsf` is cheap, only single instruction ...

Reply:
@ledvinap I didn't realise fabsf was so cheap!  Thanks.
I'm not clear on exactly what you're suggesting that we should do here.  We use `rcCommandDeltaAbs` in four places, and calculate it once.  We set the static previous abs value (ie `prevRcCommandDeltaAbs[axis]`), from the already calculated `rcCommandDeltaAbs` value. Happy to make changes if you had time to let me know more explicitly what you'd recommend.

Reply:
The idea was to store prevRcCommandDelta and use `fabsf` where necessary. But is it only removing of unnecessary micro optimization, code does work both ways.